docs/source/python/compute.rst

```diff
@@ -0,0 +1,54 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. currentmodule:: pyarrow.compute
+.. _compute:
+
+=================
+Compute Functions
+=================
+
+Arrow supports logical compute operations over inputs of possibly 
```

Comment:
what does the "logical" in here mean exactly?

Reply:
So I actually lifted the meat of this section from the corresponding C++ doc:
https://arrow.apache.org/docs/cpp/compute.html
Do we want to just delete `logical` here and in the C++ page?

Reply:
cc @pitrou

Reply:
It was meant to convey that the operations are not typed a priori (you have a single "add" function for multiple possible types, for example). But perhaps that is superfluous.